lib/pages/media_view.dart

```diff
@@ -19,6 +23,10 @@ class MediaViewPage extends HookWidget {
     final showButtons = useState(true);
     final isZoomedOut = useState(true);
 
+    final currentOpacity = useState<double>(1);
```

Comment:
opacity and position are invariants, you can compute opacity from position at build time instead of storing it (causing sometimes unnecessary re-renders)